engine/access/rpc/engine.go

```diff
@@ -141,6 +140,23 @@ func New(log zerolog.Logger,
 	return eng
 }
 
+// interceptors creates all the GRPC server interceptors
+func interceptors(rpcMetricsEnabled bool, log zerolog.Logger) grpc.ServerOption {
+
+	rateLimitInterceptor := NewRateLimiterInterceptor(log).unaryServerInterceptor
+
+	if !rpcMetricsEnabled {
+		return grpc.UnaryInterceptor(rateLimitInterceptor)
+	}
+
+	// create a chained unary interceptor
+	// first add the grpc metrics interceptor, then add the rate limit interceptor
+	// grpc metrics will report grpc API metrics before being rate limited
+	unaryInterceptors := grpc.ChainUnaryInterceptor(grpc_prometheus.UnaryServerInterceptor,
+		rateLimitInterceptor)
```

Comment:
order matters in a chain interceptor.
So on Grafana we will see all incoming request. Some of them may be rate limited.
Do you think I should add a metric to show rate limited requests?

Reply:
yup! I realized that the response code for the rejected response would be `codes.ResourceExhausted` which is unique and won't be returned otherwise. Hence, I think there is an easy way to discern which request were rejected without the need of a new metric.